monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -2703,6 +2688,21 @@ abstract class Observable[+A] extends Serializable { self =>
     new OnErrorRetryCountedObservable(self, maxRetries)
   }
 
+  /** Returns an Observable that mirrors the behavior of the source,
```

Comment:
I would mention "resetting" aspect of the retry function and add one example from backoff strategy illustrations